lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/dllmain.cpp

```diff
@@ -1244,6 +1257,37 @@ void eos_create(EOSConfig& eosConfig)
         steam_platform.SteamMajorVersion = eos_steam_config.steamSDKMajorVersion;
         steam_platform.SteamMinorVersion = eos_steam_config.steamSDKMinorVersion;
 
+        // For each element in the array (each of which is a string of an api version information)
+        // iterate across each character, and at the end of a string add a null terminator \0
+        // then add one more null terminator at the end of the array
+        std::vector<char> steamApiInterfaceVersionsAsCharArray = std::vector<char>();
+
+        // steam_platform needs to have a count of how many bytes the "array" is, stored in SteamApiInterfaceVersionsArrayBytes
+        // This has some fuzzy behavior; if you set it to 0 or count it up properly, there won't be a logged problem
+        // if you put a non-zero amount that is insufficient, there will be an unclear logged error message
+        // HACK: For now the right method seems to be counting each char added to the array, including the null terminator
+        uint32_t totalBytes = 0;
+
+        for (int apiInterfaceVersionIndex = 0; apiInterfaceVersionIndex < eos_steam_config.steamApiInterfaceVersionsArray.size(); apiInterfaceVersionIndex++)
+        {
+            std::string currentFullValue = eos_steam_config.steamApiInterfaceVersionsArray[apiInterfaceVersionIndex];
```

Comment:
I believe you can get away with using a `const std::string&` here instead of a `std::string` so that the string doesn't get copied.

Reply:
Ah, I see! This makes it point at the value in the array instead of copying it in to memory. That makes sense, I'll utilize that.